ci/docker.go

```diff
@@ -50,9 +50,19 @@ func (d *Docker) Run(ctx context.Context, job *Job, user string) (string, error)
 	// will wait until the container stops
 	waitc, errc := cli.ContainerWait(ctx, resp.ID, container.WaitConditionNotRunning)
 
+	if timeout < 1 {
+		timeout = containerTimeout
+	}
+
 	select {
 	case wErr := <-errc:
+		fmt.Println("wErr: ", wErr.Error())
 		return "", wErr
+		// if the container still running after predefined time interval, force kill it
+	case <-time.After(timeout):
+		fmt.Println("Timeout done, stopping container: ", user)
+		cli.ContainerKill(ctx, resp.ID, "SIGTERM")
+		return fmt.Sprintf("Container timed out after %d minutes", timeout), nil
```

Comment:
The `time.Duration` type has a `String()` method, and so if you print with `%v` or similar, it should print with the appropriate unit, since the timeout might not be in minutes...

Reply:
Noted.